src/background/sidebar-injector.js

```diff
@@ -438,4 +437,39 @@ export function SidebarInjector() {
       args: [clientConfig],
     });
   }
+
+  /**
+   * Injects the client's boot script, only if the config has been loaded first.
+   * If config is not found, the boot script execution is aborted.
+   * This could happen in case the config is loaded, and other script causes a navigation in the tab before the boot
+   * script loads.
+   *
+   * @param {number} tabId
+   * @param {number} [frameId]
+   */
+  async function executeClientBootScript(tabId, frameId) {
+    const configFound = await configExists(tabId, frameId);
+    if (! configFound) {
+      return Promise.resolve();
+    }
```

Comment:
These two steps `await configExits(...)` and `executeScript(...)` are separate commands sent to the process where the page is displayed, so it is possible for navigations to happen in-between.